packages/rspack-dev-server/src/patch.ts

```diff
@@ -0,0 +1,34 @@
+import WebpackDevServer from "webpack-dev-server";
+
+let old: InstanceType<typeof WebpackDevServer>["sendStats"] | undefined;
+
+function restoreDevServerPatch() {
+	// @ts-expect-error private API
+	WebpackDevServer.prototype.sendStats = old;
+}
+
+// Patch webpack-dev-server to prevent it from failing to send stats.
+// See https://github.com/web-infra-dev/rspack/pull/4028 for details.
+function applyDevServerPatch() {
+	if (old) return restoreDevServerPatch;
+
+	// @ts-expect-error private API
+	old = WebpackDevServer.prototype.sendStats;
+
+	// @ts-expect-error private API
+	WebpackDevServer.prototype.sendStats = function sendStats__rspack_patched(
```

Comment:
RspackDevServer extends WebpackDevServer to overwrite some function, this can be done in the same way, but I'm also ok with the patch way

Reply:
I'm thinking of exposing this in the future. Further discussions are required.